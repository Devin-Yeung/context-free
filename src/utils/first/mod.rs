use bnf::Term;
use std::collections::{HashMap, HashSet};

mod builder;

pub struct First<'grammar> {
    pub(crate) first: HashMap<&'grammar Term, HashSet<&'grammar Term>>,
}
