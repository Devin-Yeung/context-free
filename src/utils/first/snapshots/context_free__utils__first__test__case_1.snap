---
source: src/utils/first/mod.rs
expression: first.tabled()
---
+------+-----------+
| Term | First(X)  |
+------+-----------+
| <E>  | "(", "id" |
+------+-----------+
| <E'> | "+", "ε"  |
+------+-----------+
| <P>  | "∀", "∃"  |
+------+-----------+
| <Q>  | "∀", "∃"  |
+------+-----------+
| <R>  | "(", "id" |
+------+-----------+
| <T>  | "(", "id" |
+------+-----------+
